### DISCUSSION TITLE
Running dirinstall in a container?

### DISCUSSION THREAD

**Discussion by tofraa** [rhinstaller/anaconda#5849] [Q&A]
I´ve been using anaconda with --dirinstall for almost as long as it has supported it, and it works great. Recently I have been trying to get a basic command: "anaconda --ks minimal.ks --cmdline --dirinstall" to work. The storage module fails to initialise, with blivet complaining about missing default filesystem type (understandably since it's running in a container). I have been able to patch blivet into returning True at various places, but this is not really a sustainable solution

Disabling the storage module turns into a non-ending stream of errors, some which I have found a work-around for by changing the anaconda configuration. There seems to be a requirement for the storage module to be present for most initialisation that anaconda f.eks  Storage Contraints seems to be using the storage module. 

I far as I can see a dir-install should not require the storage module, but relies unconditionally on its initialisation anyway.

Is there a way to get anaconda to run in dirinstall inside a container?

Can the storage modules be disable, or at least set as optional, during a dirinstall?

One can make the argument that this is a blivet is at fault here, but I am not sure if running blivet inside a container is really that useful at all. Perhaps a solution might be to alter the storage module to skip blivet initialisation if it detects an unusable environment?

**Comment by jkonecny12** [comment]
Hi, I think this won't be that easy to do. Other modules are using storage module for information about devices which unfortunately is for example happening in payload when we are looking for installation source. In general it shouldn't be required but I don't think this is easily achievable.
